autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectColumn.tsx

```diff
@@ -17,6 +17,22 @@ const CourseSelectColumn: React.FC = () => {
 
   const dispatch = useDispatch();
 
+  const term = useSelector<RootState, string>((state) => state.term);
+
+  // When term is changed, fetch saved courses for the new term
+  React.useEffect(() => {
+    if (!term) return;
+    fetch(`sessions/get_saved_courses?term=${term}`).then((res) => (
+      res.json()
+    )).then((courses: SerializedCourseCardOptions[]) => {
+      dispatch(replaceCourseCards(courses, term));
+    }).catch(() => {
+      // If unable to parse saved course cards or request fails, go back to default
+      // of one empty course card
+      dispatch(replaceCourseCards([], term));
```

Comment:
Is there a difference between this and `clearCourseCards()`?

Reply:
`replaceCourseCards` with an empty array creates the initial empty card, now that I look back at this I think I could just make the initial state a single course card again, and then if there are saved course cards it should be able to correctly update the initial blank one and avoid ambiguity like this